src/ripple/basics/BasicConfig.h

```diff
@@ -76,6 +76,35 @@ class Section
         return values_;
     }
 
+    /**
+     * Set the legacy value for this section.
+     */ 
+    void
+    legacy(std::string value)
+    {
+        if (lines_.empty())
+            lines_.emplace_back(std::move(value));
+        else
+            lines_[0] = std::move(value);
+    }
+
+    /**
+     * Get the legacy value for this section.
+     *
+     * @return The retrieved value. A section with an empty legacy value returns
+               an empty string.
+     */
+    std::string
+    legacy() const
+    {
+        if (lines_.empty ())
+            return "";
+        else if (lines_.size () > 1)
```

Comment:
Good observation! The `lines_` variable can be filled by parsing a config file, and if you try to access it that this code path is triggered. So code is correct as written (I even have a unit test for this case): https://github.com/seelabs/rippled/blob/basicConfigLegacy/src/ripple/core/tests/Config.test.cpp#L271-272